contract/services.py

```diff
@@ -1,41 +1,157 @@
+                "save": False,
+            }
+        )
+        return result_contract_valuation["data"]["total_amount"]
 
     def submit(self, submit):
         pass
 
     def amend(self, submit):
         pass
 
-    def create(self, contract):
-        pass
-
     def update(self, contract):
         pass
 
     def delete(self, contract):
         pass
 
 
-@core.comparable
-class PolicyHolderContractDetails(object):
-
-    def __init__(self, policy_holder):
-        self.policy_holder = policy_holder
-        pass
-
-    def __eq__(self, other):
-        return isinstance(other, self.__class__) and self.__dict__ == other.__dict__
+class ContractDetails(object):
+
+    def __init__(self, user):
+        self.user = user
+
+    @check_authentication
+    def update_from_ph_insuree(self, contract_details):
+        try:
+            contract_insuree_list = []
+            policy_holder_insuree = PolicyHolderInsuree.objects.filter(policy_holder__id=contract_details['policy_holder_id'])
+            for phi in policy_holder_insuree:
+                cd = ContractDetailsModel(
+                    **{
+                        "contract_id": contract_details["contract_id"],
+                        "insuree_id": phi.insuree.id,
+                        "contribution_plan_bundle_id": str(phi.contribution_plan_bundle.id),
+                    }
+                )
+                cd.save(self.user)
+                uuid_string = str(cd.id)
+                dict_representation = model_to_dict(cd)
+                dict_representation["id"], dict_representation["uuid"] = (str(uuid_string), str(uuid_string))
+                dict_representation["policy_id"] = phi.last_policy.id
+                contract_insuree_list.append(dict_representation)
+        except Exception as exc:
+            return _output_exception(model_name="ContractDetails", method="updateFromPHInsuree", exception=exc)
+        return _output_result_success(dict_representation=contract_insuree_list)
+
+
+class ContractContributionPlanDetails(object):
+
+    def __init__(self, user):
+        self.user = user
+
+    @check_authentication
+    def contract_valuation(self, contract_contribution_plan_details):
+        try:
+            dict_representation = {}
+            total_amount = 0
+            for contract_details in contract_contribution_plan_details["contract_details"]:
+                cp = ContributionPlanBundleDetails.objects.filter(
+                    contribution_plan_bundle__id=str(contract_details["contribution_plan_bundle"])
+                )[0]
+                ccpd = ContractContributionPlanDetailsModel(
+                    **{
+                        "contract_details_id": contract_details["id"],
+                        "contribution_plan_id": str(cp.id),
+                        "policy_id": contract_details["policy_id"]
+                    }
+                )
+                # temporary value until calculation module be developed
```

Comment:
Could you add "# TODO" where there is a code to update so we can checkk if all is done at the end https://www.python.org/dev/peps/pep-0350/

Reply:
Ok.